node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1426,8 +1421,8 @@ async fn execute_blocks(
                 )
                 .await?;
                 info!(block_hash=%block.hash(), "finish - re-executing finalized block");
-                if block == *block_and_execution_effects.block() {
-                    success = true;
+                blocks_match = block == *block_and_execution_effects.block();
+                if blocks_match {
```

Comment:
Maybe simpler:
* Get rid of `blocks_match`.
* Just do `while block != *block_and_execution_effects.block()`
* Put the `warn!` at the beginning of the loop. No `break` required.

Reply:
`break` will still be required to get out of the loop over the complete set of peers. We have a nested loop because that's the simplest way I could think of to iterate over the peers, but still continue if we didn't get a matching block.

We _could_ do `let mut peers = vec![]` before the `while` loop, and then in the loop do `if peers.is_empty() { get_more }; let peer = peers.pop();`, but I'm not sure if it's much better.

Reply:
Ah, right, I totally missed the second loop! :facepalm: Then it's fine as it is.